### PR TITLE
Fix `declaration-property-value-no-unknown` false negatives for `position-try-options`

### LINKED ISSUE #7964: Fix `declaration-property-value-no-unknown` false negatives for `position-try-options`
### What minimal example or steps are needed to reproduce the bug?

```css
a {
  position-try: foo;
  position-try-fallbacks: foo;
  position-try-options: foo;
  position-try-order: foo;
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": { "declaration-property-value-no-unknown": true }
}
```

### How did you run Stylelint?

[demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdilAHCAzgJYAuREmAtCQE4Ce8KAZhBANxY77FkXX2UmSADbCARkjABrAoxbtOeQqXJVadShFy9Ms5qw7YlPVfw0QasGnINYAviAA0IJkWEwAckgC2cRDAAPH1x3ADowAgIncApXAHMEEED8GhIUWCEAV2E0jCN0EBpsmAICxmAUAtgwYSQaJB1KXBotGFSNADcRTJhKTAhKTMwpfoB3TDKUWh6UO3ssaMhMeIAxC28GxIISOndhIkwScNiiONDvACso51hcKMQ8nALt3Zh9w8mC2pISkgLHRTPHZ7A4kShLeKUbZITBQOpQT4gb6-Ar2EB2IA)

### What did you expect to happen?

`Unexpected unknown value "foo" for property "position-try-options"`

### What actually happened?

`✅ No problems!`

### Resources

mdn/browser-compat-data#23651 w3c/csswg-drafts#10395
https://caniuse.com/?search=position-try

### PR DESCRIPTION
**Pull request by Mouvedia**
> Which issue, if any, is this issue related to?

Closes #7964

> Is there anything in the PR that needs further explanation?

While waiting for #8100 we can fix some of the false positives/negatives.

https://caniuse.com/mdn-css_properties_position-try-fallbacks

see also #8382

### CODE REVIEW

File: lib/rules/declaration-property-value-no-unknown/index.mjs [hunk 1]
```diff
@@ -69,6 +69,9 @@ const rule = (primary, secondaryOptions) => {
 
 		/** @type {SecondaryOptions['propertiesSyntax']} */
 		const propertiesSyntax = {
+			'overflow-x': '| <-non-standard-overflow>',
+			'overflow-y': '| <-non-standard-overflow>',
+			'position-try-options': "<'position-try-fallbacks'>",
```
Comment (ryo-manba) [inline lib/rules/declaration-property-value-no-unknown/index.mjs:74]: Based on [this discussion](https://github.com/stylelint/stylelint/issues/7964#issuecomment-2351588284), Stylelint shouldn't actively support removed or deprecated CSS features unless they have a broad impact. I think this change shouldn't be made. What do you think?

File: lib/rules/declaration-property-value-no-unknown/__tests__/index.mjs [hunk 2]
```diff
@@ -62,6 +62,9 @@ testRule({
 		{
 			code: 'a { overflow: overlay; }',
 		},
+		{
+			code: 'a { overflow-y: overlay; }',
```
Comment (ryo-manba) [inline lib/rules/declaration-property-value-no-unknown/__tests__/index.mjs:66]: [minor]
I  think it would be good to include a test for `overflow-x` as well since it overrides the syntax.

File: lib/rules/declaration-property-value-no-unknown/index.mjs (hunk 1 above)
Comment (Mouvedia) [reply lib/rules/declaration-property-value-no-unknown/index.mjs:74]: I think it's harmless and it will be discussed more seriously once we will transition to #8100.
i.e. #8100 doesn't plan to support that kind of renames

File: lib/rules/declaration-property-value-no-unknown/__tests__/index.mjs (hunk 2 above)
Comment (Mouvedia) [reply lib/rules/declaration-property-value-no-unknown/__tests__/index.mjs:66]:
```suggestion
			code: 'a { overflow-x: -moz-scrollbars-none; overflow-y: overlay; }',
```

File: lib/rules/declaration-property-value-no-unknown/index.mjs (hunk 1 above)
Comment (ryo-manba) [reply lib/rules/declaration-property-value-no-unknown/index.mjs:74]: As far as I know, extending already renamed features like this hasn't been done before.
Adding it here would set a precedent.

Since multiple members have expressed opposition to this change, I think we should avoid adding it.
I’m also against it. If needed, users can easily extend it themselves.

Comment (Mouvedia) [reply lib/rules/declaration-property-value-no-unknown/index.mjs:74]: What I am doing for `overflow-x` and `overflow-y` is similar; these values are non-standard but are or used to be supported by browsers.
i.e. either we accept the PR as a whole on the same principle or we don't ever support anything "non-standard" and close the PR

Comment (ryo-manba) [reply lib/rules/declaration-property-value-no-unknown/index.mjs:74]: Sorry for the late reply.
I submitted a PR to CSSTree to handle these properties upstream:

- https://github.com/csstree/csstree/pull/324
- https://github.com/csstree/csstree/pull/323

This was based on [Romain's comment](https://github.com/stylelint/stylelint/issues/7964#issuecomment-2351588284):

> (You could open an issue upstream with either csstree or mdn-data to request that they preserve position-try-options as valid.)

Let me know what you think.

Comment (Mouvedia) [reply lib/rules/declaration-property-value-no-unknown/index.mjs:74]: That's great but since none of us have write access on csstree, it might take a while.
Let's wait and see.

**Comment by Mouvedia** [comment]
@ybiquitous both @ryo-manba's PRs are still unmerged; should we move on with _this_ PR or wait for the resolution of #8465?

**Comment by ybiquitous** [comment]
Users can avoid the issues by using the `propertiesSyntax` secondary option, right? Does Stylelint or CSSTree need to continue supporting legacy properties?

**Comment by Mouvedia** [comment]
> supporting legacy properties

If you meant vendor-prefixed values and deprecated values, stylelint distinguishes between known—i.e. ever supported—and deprecated since we added `declaration-property-value-keyword-no-deprecated` rule. For vendor-prefixed values, [csstree already does for `overflow`](https://csstree.github.io/docs/syntax/#Property:overflow) so it makes perfect sense to extend it to `overflow-x`/`overflow-y` as well—i.e. it was probably an oversight.

My question was not about the merit of this PR but about whether we continue to wait or take action.

**Comment by ybiquitous** [comment]
I'm in favor of waiting unless this PR is not requested by many users.

**Comment by romainmenke** [comment]
See: https://github.com/stylelint/stylelint/issues/8339#issuecomment-3591469508

**Comment by Mouvedia** [comment]
csstree/csstree#324 has been merged but we are still waiting for a new release of `css-tree`; let's keep this PR open as a reminder until then.

**Comment by romainmenke** [comment]
`css-tree` was released and this issue was resolved: [Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdilAHCAzgJYAuREmAtCQE4CelEuZFB8KAZhBANxYC+IADQgORADYwAckgC2cRDAAec3JIB0YAgWHgKYgOYIQy-DRIpYHJAFdxFjNhToQNOzAIv2wZyFhhxJBokFipcGiYYcwYANyRxGxhKTAhKG0wAaxSAd0wvFFpElH4BLF1ITEMAMQgaWRDjAhI6SXEiTBJNfSIDdVkAKx0RWFwdREccFyaWmDaO-JdAkg8SFyEsSZBp1vaSSgrDSiakTCggqAWQJZWXARB+IA)
